### PR TITLE
add economic order quantity model

### PR DESCRIPTION
**Pull request by peisenha**
We want to add the economic order quantity model.

- [x] first draft
- [x] add tests of function
- [x] polish references citations

### CODE REVIEW

**Comment by peisenha** [comment]
@loikein , please work with @MargaritaRad  that we have proper citation style in https://github.com/OpenSourceEconomics/temfpy/blob/534997bbf88a3791c34f3b18520bb60adf3b186f/temfpy/uncertainty_quantification.py#L76

I will set up a proper testing infrastructure. No need to hurry, just let me know when fixed.

**Comment by peisenha** [comment]
@loikein , I am now also done with setting up proper testing infrastucture. Please just fix the references and then let's try to get this merged before the check-in on Wednesday. Just assign it ot me for review once you are done.

**Comment by loikein** [comment]
@peisenha Great, I will start working…

**Comment by loikein** [comment]
Ok, but what can (?) I do…

```
 ======================= 18 passed, 15 warnings in 2.08s ========================
./temfpy/uncertainty_quantification.py:79:101: E501 line too long (101 > 100 characters)
./temfpy/uncertainty_quantification.py:81:101: E501 line too long (130 > 100 characters)
./temfpy/uncertainty_quantification.py:83:101: E501 line too long (164 > 100 characters)
##[error]Process completed with exit code 1.
```

**Comment by loikein** [comment]
@peisenha This error is hard to fix. If I write:

```rst
.. [3] Economic order quantity.
    (2020, April 3).
    In Wikipedia.
    Retrieved from https://en.wikipedia.org/w/index.php
    ?title=Economic_order_quantity&oldid=948881557
```

then Sphinx will render as: (the URL is broken)

Economic order quantity. (2020, April 3). In Wikipedia. Retrieved from https://en.wikipedia.org/w/index.php ?title=Economic_order_quantity&oldid=948881557

**Comment by loikein** [comment]
And then, if I write according to [Stack Overflow](https://stackoverflow.com/questions/14414878/how-do-i-break-a-link-in-a-rst-docstring-to-satisfy-pep8):

```rst
.. [3] Economic order quantity.
    (2020, April 3).
    In Wikipedia.
    Retrieved from 
https://en.wikipedia.org/w/index.php\
    ?title=Economic_order_quantity&oldid=9
48881557
```

the link has correct appearance, but does not has correct HTML structure:

图片

**Comment by loikein** [comment]
Huge thank you to [bgme](https://bgme.me/@bgme/104207149151655236) who suggested the neat solution.

File: temfpy/uncertainty_quantification.py [hunk 1]
```diff
@@ -1,38 +1,122 @@
...
+    -----
+
+    A historical perspective on the model is provided by [2]_. A brief description with the core
+    equations is available in [3]_.
+
+    References
+    ----------
+
+    .. [1] Harris, F. W. (1990).
+        How Many Parts to Make at Once.
```
Comment (peisenha) [inline temfpy/uncertainty_quantification.py:80]: We are not consistent here with the capitalization of the titles. Here we have all capitalized but in [3] it is not. Please follow instructions here: https://github.com/OpenSourceEconomics/bibliography  If that does not have all the information you need than that is our fault and please work with @luward  to expand the description accordingly.

File: temfpy/uncertainty_quantification.py [hunk 2]
```diff
@@ -1,38 +1,122 @@
...
+
+    rslt = (1 + b * x[2] ** 4) * np.sin(x[0]) + a * np.sin(x[1]) ** 2
+    return rslt
+
+
+def eoq_harris(x, r=10):
+    r"""Economic order quantity model.
+
+    This function computes the optimal economic order quantity (EOQ) based on the model presented in
+    [1]_. The EOQ minimizes the holding costs as well as ordering costs. The core parameters of
```
Comment (peisenha) [inline temfpy/uncertainty_quantification.py:51]: Screenshot from 2020-05-22 10-42-00

The numbers show up kind of weird in the compiled version, see attached. Is there any configurations we can set that they look normal, i.e. just a simple `[1]` on the line now they are lookig like a superscript. I know that this was already the case when I handed it over to you, but please see if you can find a better solution.

File: temfpy/uncertainty_quantification.py [hunk 3]
```diff
@@ -1,38 +1,122 @@
...
+
+    m, s, c = x
+    y = np.sqrt((24 * r * m * s) / c)
+
+    return y
+
+
+def simple_linear_function(x):
+    """Simple linear function.
+
```
Comment (peisenha) [inline temfpy/uncertainty_quantification.py:109]: I forgot to add a brief description here that we actually take the sum of all elements. Please add.

File: temfpy/uncertainty_quantification.py (hunk 2 above)
Comment (loikein) [reply temfpy/uncertainty_quantification.py:51]: I think this is because one reference cannot be referred to twice…? Let me try to fix it.

Comment (loikein) [reply temfpy/uncertainty_quantification.py:51]: Yes, yes.

File: temfpy/uncertainty_quantification.py (hunk 3 above)
Comment (loikein) [reply temfpy/uncertainty_quantification.py:109]: But, why do we need this function… If it's just `sum()`…

Comment (peisenha) [reply temfpy/uncertainty_quantification.py:110]: I want us to be able to iterate simply over all test functions in the UQ literature later.  Please always follow the docstring guide and include a summary line in the beginning ... regardless of how small our function is :). I will add this now, so we can merge.

File: temfpy/uncertainty_quantification.py (hunk 2 above)
Comment (peisenha) [reply temfpy/uncertainty_quantification.py]: Great, thanks!

File: temfpy/uncertainty_quantification.py (hunk 1 above)
Comment (peisenha) [reply temfpy/uncertainty_quantification.py:80]: If I raise an issue, please do not mark as resolved yourself, just comment that you addressed it and then I can look at it an mark as resolved. Why is now all capitalized? This is not in line with description in our bib repository? Let me know if I am missing sth. If we need to change it here, please also adjust in the resources section. Regarding the superscript issue, great, thanks!

File: temfpy/uncertainty_quantification.py (hunk 3 above)
Comment (loikein) [reply temfpy/uncertainty_quantification.py:114]: I have added it back, and also for other functions that does not contain an one-line summary.

**Pull request merged**